app/Http/Middleware/Google2FA.php

```diff
@@ -15,10 +17,28 @@ class Google2FA
     public function handle($request, Closure $next)
     {
         $authenticator = app(Authenticator::class)->boot($request);
+        $user = $request->user();
 
-        if ($authenticator->isAuthenticated()) {
-            return $next($request);
+        if ($user !== null) {
+            $remember = $request->cookie($user->getRememberDeviceKey());
+
+            // If tokens do not match, cookie is no longer valid.
+            if ($remember !== null && $remember !== $user->getRememberDeviceToken()) {
+                Cookie::forget($user->getRememberDeviceKey());
+            }
+
+            if ($authenticator->isAuthenticated() || ($remember !== null && $remember === $user->getRememberDeviceToken())) {
+                if (!$authenticator->isAuthenticated()) {
+                    Log::notice('User skipped OTP entry with known device.', ['id' => $request->user()->id]);
```

Comment:
I think we should call $authenticator->login() at this point. Otherwise, this message will be logged with every single request, and we'll have to add these same token checks to our Require2FA middleware.

Reply:
Ah good point, I'll do that